eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -321,6 +321,8 @@ jobs:
           # Allows Arcade to have access to the commit for the build, pass it through to the container
           customEnvVars="$customEnvVars BUILD_SOURCEVERSION=$BUILD_SOURCEVERSION"
 
+          customEnvVars="$customEnvVars BUILD_BUILDNUMBER=$(Build.BuildNumber)"
```

Comment:
Depending on how the manifest metadata is determined, you may eventually need to add more variables, but I think this is fine for now: https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L202-L209